core/utils/utils.go

```diff
@@ -745,3 +745,13 @@ func (t *ResettableTimer) Reset(duration time.Duration) {
 	}
 	t.timer = time.NewTimer(duration)
 }
+
+func BytesToUint64(buf []byte) uint64 {
```

Comment:
Would it be better to use `[4]byte` here?

Reply:
We don't have a guaranteed size of `dataLengthBytes`. We could trim/expand it, but this seemed simpler.